healthcare/healthcare/doctype/patient_appointment/patient_appointment.py

```
@@ -460,6 +462,78 @@ def update_event(self):
 
+	def _get_token_lock_name(self):
+		scope_parts = [
+			str(self.appointment_date or ""),
+			str(self.appointment_for or ""),
+			str(self.practitioner or ""),
+			str(self.appointment_time or ""),
+			str(self.service_unit or ""),
+			str(self.department or ""),
+		]
+		raw_key = "|".join(scope_parts)
+		key_hash = hashlib.sha1(raw_key.encode()).hexdigest()
```

Ruff (0.15.4)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)